services/121-service/src/fsp-integrations/account-management/cooperative-bank-of-oromia-account-management/cooperative-bank-of-oromia-account-management.service.ts

```diff
@@ -0,0 +1,138 @@
+import { MappedPaginatedRegistrationDto } from '@121-service/src/registration/dto/mapped-paginated-registration.dto';
+import { RegistrationViewScopedRepository } from '@121-service/src/registration/repositories/registration-view-scoped.repository';
+import { RegistrationsPaginationService } from '@121-service/src/registration/services/registrations-pagination.service';
+
+@Injectable()
+export class CooperativeBankOfOromiaAccountManagementService {
+  constructor(
+    private readonly cooperativeBankOfOromiaAccountValidationScopedRepository: CooperativeBankOfOromiaAccountValidationScopedRepository,
+    private readonly registrationViewScopedRepository: RegistrationViewScopedRepository,
+    private readonly registrationsPaginationService: RegistrationsPaginationService,
+    private readonly cooperativeBankOfOromiaService: CooperativeBankOfOromiaService,
+    private readonly programRepository: ProgramRepository,
+  ) {}
+
+  public async retrieveAndUpsertAccountValidations(): Promise<number> {
+    const programs = await this.programRepository.getAllProgramIdsWithFsp(
+      Fsps.cooperativeBankOfOromia,
+    );
+    let totalAccountValidations = 0;
+    for (const programId of programs) {
+      const accountValidationsPerProgram =
+        await this.retrieveAndUpsertAccountValidationsForProgram({ programId });
+      totalAccountValidations += accountValidationsPerProgram;
+    }
+    return totalAccountValidations;
+  }
+
+  public async retrieveAndUpsertAccountValidationsForProgram({
+    programId,
+  }: {
+    programId: number;
+  }): Promise<number> {
+    const queryBuilder =
+      this.registrationViewScopedRepository.getQueryBuilderFilterByFsp({
+        programId,
+        fspName: Fsps.cooperativeBankOfOromia,
+      });
+    const registrationsWithCoopBank =
+      await this.registrationsPaginationService.getRegistrationViewsNoLimit({
+        programId,
+        paginateQuery: {
+          path: '',
+          select: [
+            'id',
+            FspAttributes.fullName,
+            FspAttributes.bankAccountNumber,
+          ],
+        },
+        queryBuilder,
+      });
+    for (const registration of registrationsWithCoopBank) {
+      const accountInformation =
+        await this.cooperativeBankOfOromiaService.getAccountInformation(
+          registration[FspAttributes.bankAccountNumber]!, // all registrations with coop bank have account number
+        );
+      await this.storeAccountValidationResult(registration, accountInformation);
+    }
+
+    return registrationsWithCoopBank.length;
+  }
+
+  private async storeAccountValidationResult(
+    registration: MappedPaginatedRegistrationDto,
+    accountInformation: {
+      cooperativeBankOfOromiaName?: string;
+      errorMessage?: string;
+    },
+  ): Promise<void> {
+    const namesMatch = this.doTheNamesMatch({
+      registrationName: registration[FspAttributes.fullName]!,
+      bankName: accountInformation.cooperativeBankOfOromiaName,
```

Comment:
it was not immediately clear to me this was about the registrationName according to the bank. I thought it was about the name of the bank. I get it now, but can maybe be clear (no immediate suggestion).